### DISCUSSION TITLE
github copilot keeps showing "loading solutions" vs2022

### DISCUSSION THREAD

**Discussion by PGTBoos** [community/community#31929] [Copilot Conversations]
As I have an interest in AI I decided to join the copilot program to see what it is able to do.
I installed it in visual studio 2022, it does do a few suggestions of code and indeed reduces boilerplate coding time.

However, in VS2022 in its own panel it keeps showing "loading solutions" and I wonder if this is normal, 
As a paying customer, I don't know who to contact, but I hope this finds the right person or someone else who knows more about it.

**Comment by baodden** [comment]
I have the same problem with both the Community and Professional version of VS 2022. No apparent fix to get the inline and solutions tabs to start working again, other than restarting the whole IDE. No kind of error showing in the output tab either.

**Reply by RyanCannady** [reply]
Im also having this issue, I left mine "loading solutions" for around 5 minutes before restarting VS.
Seems to be the only current fix!

**Reply by padillacm1** [reply]
For me solved by disconnecting from VPN

**Reply by RyanCannady** [reply]
> For me solved by disconnecting from VPN

I don't use a VPN, won't work for me unfortunately.
Weird that we are all having this bug happen around the same time, must've  been an update or something of the sort.

**Reply by andriigentech** [reply]
Same issue. It shows first suggestion, and stay in Loading Solutions state until VS restart
version: 1.43.01
![image](https://user-images.githubusercontent.com/113241303/189485942-32649e3f-3dc7-44e5-9d50-1ae3074eadb8.png)

**Reply by tilmann1** [reply]
The engineering team is currently investigating a regression with VS 17.3.x, if you can please try using Copilot 1.43.0.1 with VS 17.2.x which should not cause completions to hang after a while. Thanks!

**Reply by tilmann1** [reply]
Please try the just released Copilot 1.44.0.0 VS extension which contains various improvements that help address this issue on VS 17.3.x. Thanks!

**Reply by cmp-nct** [reply]
Is there any update ?
For me Copilot works on Jetbrains but it permanently crashes/hangs like this in VS

**Reply by tilmann1** [reply]
@cmp-nct Which version of Visual Studio and which version of the Copilot VS extension are you on?

With VS 17.3.x and Copilot VS extension 1.56.0.2 there should be no permanent crashes/hangs.

**Reply by cmp-nct** [reply]
> @cmp-nct Which version of Visual Studio and which version of the Copilot VS extension are you on?
> 
> With VS 17.3.x and Copilot VS extension 1.56.0.2 there should be no permanent crashes/hangs.
I'm on latest version of copilot and latest version of VS (17.4.0)

If I had started out with Copilot on Visual Studio I would have canceled the subscription after 30 minutes.
It works horrible bad there.
It not only hangs 3-5 times per hour needing an IDE restart, it also keeps stopping to suggest in the IDE needing to manually open the "panel" all the time.
When you click on "accept solution" it doesn't accept it but moves your cursor into the IDE where you look at a totally different solution suggested, you then need to cycle through them as if you never accepted one.
It keeps adding duplicate closing braces, mangling the source code.

**Reply by ayim** [reply]
@cmp-nct, @baodden, @RyanCannady, @andriigentech, @padillacm1 we shipped a fix in Copilot v1.78.0.0 that fixes this issue.
Two easy steps to update:
- In Visual Studio, go to Extensions > Manage Extensions > Updates and click Update if available
- In the same dialog, click Installed, and make sure "Automatically update this extension" is checked for GitHub Copilot

**Comment by atomisadev** [comment]
Duplicate of #13491

It is suggested that you update to the latest version of GitHub Copilot.

**Reply by baodden** [reply]
No, I'm on 1.43.0.0 of the Copilot extension, and Visual Studio 2022 17.3.3, both being the latest available, and the issue still stands.

**Reply by PGTBoos** [reply]
indeed i'm subscribed (and a paying customer) to and receive the latest updates automatically,, this question is therefore not a duplicate. it's on version 1.43.00

**Reply by doanamo** [reply]
Same problem for on 1.43.0.0 and Visual Studio 2022 17.3.3

**Reply by SenMorgan** [reply]
I have the same problem last few days. VS version 17.3.3, Copilot version 1.43.0.0.
But in VS Code on the same PC everything is OK.

**Reply by RyanCannady** [reply]
Im having the same issue and I just installed GitHub Copilot...
Pretty sure it's up to date on my end!

**Reply by PGTBoos** [reply]
Although its own panel works, i've noticed that restarting VS2022 restarts copilot and then it works again for a few minutes, after which it becomes non-responsive again (pig keeps rotating no suggestions), (my internet connection is perfect and not to blame).
I have no idea on how to restart copilot only, without restarting VS2022.

Does anyone know if this forum is the way to contact Copilot ?, after all we pay for a service that doesn't work, normally you get into a ticketing system for support, but its hard to find them, or are they aware of this thread ?.

**Reply by RyanCannady** [reply]
@PGTBoos  I emailed them about it, I will let you know if I hear anything back!

**Reply by tilmann1** [reply]
The engineering team is currently investigating a regression with VS 17.3.x, if you can please try using Copilot 1.43.0.1 with VS 17.2.x which should not cause completions to hang after a while. Thanks!

**Reply by tilmann1** [reply]
Please try the just released Copilot 1.44.0.0 VS extension which contains various improvements that help address this issue on VS 17.3.x. Thanks!

**Reply by RyanCannady** [reply]
> Please try the just released Copilot 1.44.0.0 VS extension which contains various improvements that help address this issue on VS 17.3.x. Thanks!

Will do, thank you!

**Reply by PGTBoos** [reply]
Well I updated but don't see the difference, I now have my .cs file open in VSCODE and Visual Studio.
It works great in VSCODE, with frequent hints, but VS2022 I get nothing (it's up to date 1.44.0.0), 
It should just work in VS2022 too, 

my errors :

[Error] An error occurred for 'OnCompletionElapsed': StreamJsonRpc.RemoteInvocationException: Request getCompletions failed with message: The operation was aborted
   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__143`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotClient.<RunOperation>d__13`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotService.<RunOperation>d__7`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotSuggestionProvider.<TriggerInlineSuggestionAsync>d__35.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotCommandHandler.<TriggerInlineSuggestionAsync>d__35.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotCommandHandler.<<OnCompletionElapsed>b__8_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotCommandHandler.<>c__DisplayClass34_0.<<ExecuteCommandCore>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Shell.VsTaskLibraryHelper.<>c__DisplayClass35_0.<<FileAndForget>b__0>d.MoveNext()
[Error] An error occurred for 'OnCompletionElapsed': StreamJsonRpc.RemoteInvocationException: Request getCompletions failed with message: The operation was aborted.
   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__143`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotClient.<RunOperation>d__13`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotService.<RunOperation>d__7`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotSuggestionProvider.<TriggerInlineSuggestionAsync>d__35.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotCommandHandler.<TriggerInlineSuggestionAsync>d__35.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotCommandHandler.<<OnCompletionElapsed>b__8_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotCommandHandler.<>c__DisplayClass34_0.<<ExecuteCommandCore>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Shell.VsTaskLibraryHelper.<>c__DisplayClass35_0.<<FileAndForget>b__0>d.MoveNext()
[Error] An error occurred for 'BackspaceKeyCommandArgs': StreamJsonRpc.RemoteInvocationException: Request getCompletions failed with message: The operation was aborted.
   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__143`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotClient.<RunOperation>d__13`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotService.<RunOperation>d__7`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotSuggestionProvider.<TriggerInlineSuggestionAsync>d__35.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotCommandHandler.<TriggerInlineSuggestionAsync>d__35.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotCommandHandler.<>c__DisplayClass3_0.<<ExecuteCommand>b__2>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotCommandHandler.<>c__DisplayClass34_0.<<ExecuteCommandCore>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Shell.VsTaskLibraryHelper.<>c__DisplayClass35_0.<<FileAndForget>b__0>d.MoveNext()

**Reply by Therealkorris** [reply]
Exactly the same here, the newest Copilot, VS2022 Professional, but it works in VSCode and VS Community.
Nothing happens when I restart the IDE either. Signed in and out, reinstalled copilot and VS

**Comment by cmp-nct** [comment]
I've had it crash in so many ways on VS.
Just restart the IDE, you have no other options.

**Comment by jholovacs** [comment]
I just installed this with the latest version of copilot and vs, and am seeing this issue months later. Seems like a pretty serious problem if this is going to be taken seriously...

**Reply by tilmann1** [reply]
@jholovacs Which version of VS are you on? Is it a VS 17.5 preview release? Thanks!

**Comment by jholovacs** [comment]
@tilmann1 latest stable, 17.4.4

**Reply by tilmann1** [reply]
@jholovacs Can you still reproduce this reliably with VS 17.4.4 and version 1.69 of the Copilot VS extension? Thanks!

**Comment by joost-morsink** [comment]
I'm having the same issue with VS pro 17.4.4 and Copilot 1.70. Signed out/in multiple times.

output:
GitHub Copilot 1.70.0.0 (v1.70.0.0@b8131dff7)
GitHub Copilot Agent v1.70.0
statusNotification: Warning
statusNotification: Warning
statusNotification: Warning

**Reply by K97i** [reply]
This discussion is from last year, but I might as well answer.

The entirety of Github, and by proxy, Copilot is [pretty much having a seizure right now.](https://www.githubstatus.com/incidents/9c941kssjk0q) You might have to wait this one out.

**Comment by jholovacs** [comment]
@tilmann1 I actually uninstalled copilot. It was unusable, and it interfered with VS Intellicode (and by extension my coding) too much.  I found its suggestions to be 85-95% flat wrong and obtuse and extremely disruptive to my work. It's a nice idea, but for me, I found it nowhere near ready for primetime. I've already cancelled my subscription and won't be looking at this again for some time.

**Comment by nolanblew** [comment]
I've been using it on VS Code, Android Studio, and Visual Studio 2022 community (preview and stable) since it was in beta, and it works fantastic in VS Code and Android Studio, but I've never got it to consistently work in VS22, any version. Before version ~1.30, copilot would crash after about 2 or 3 suggestions in and only restarting VS would fix it. Newest version is likely doing the same but without any logging.

Really hope this gets fixed soon. It's fantastic in VS Code and Android Studio. VS is my favorite and go-to IDE so it's sad to see that Microsoft-owned copilot doesn't even work in their premier IDE

**Reply by tilmann1** [reply]
@nolanblew Is there a specific issue with VS 17.4.4 and version 1.70 of the Copilot VS extension that the Copilot team could look into? The team is not aware of any major issues at this point and we're always looking for bug reports. Thanks!

**Comment by PGTBoos** [comment]
I wonder I'm using chatgpt now more often does the code suggestion windo the one with multiple longer answers work now in vscode Vs studio

Outlook voor Android downloaden<https://aka.ms/ghei36>
________________________________

**Reply by cmp-nct** [reply]
ChatGPT uses Davinci-003 which is significantly ahead of Davinci-002 and Codex is based on 002.
003 is not just "ahead" it's so much better you can not even compare them.

Copilot is still useful, as it saves you typing work and often can adapt snippets.
I love it's writing skill for debug output, my usual debug output is short and sucks since Copilot I have great and detailed debug messages

Though there is no question .. Copilot needs to upgrade to 003 to be competitive.

**Reply by ayim** [reply]
@cmp-nct have you seen this?
I think it's what you're looking for, and there's a public preview you can sign up for this
https://devblogs.microsoft.com/visualstudio/github-copilot-chat-for-visual-studio-2022/

**Comment by PGTBoos** [comment]
Yes copilot needs a brain update

What I like about chatgpt is to be able to as it questions and to let it explain pieces of code or to sugest code examples for example Qlearning or genetic evolution in deep neural networks and other algorithms like compression it knows a lot more advanced stuff not always the historical details like origin creater usage though in code it knows
In a few hours lots can be learned this way. It does sometimes create errors but when prompted it corrects them. In fact in some of the lightsout crashes it has good ideas on how to debug 3th party code ( python libs)


Outlook voor Android downloaden<https://aka.ms/ghei36>
________________________________

**Reply by ayim** [reply]
@PGTBoos is this the brain update you've been looking for?
https://devblogs.microsoft.com/visualstudio/github-copilot-chat-for-visual-studio-2022/

**Comment by PatrikUgrin** [comment]
Hi, this is still a problem a year after... So is copilot completely useless?

**Reply by ayim** [reply]
@Brigapes As mentioned in the other message, we shipped a fix in Copilot v1.78.0.0 that fixes this issue (hopefully for good!)
Two easy steps to update:

- In Visual Studio, go to Extensions > Manage Extensions > Updates and click Update if available
- In the same dialog, click Installed, and make sure "Automatically update this extension" is checked for GitHub Copilot

**Comment by PickleOnAString** [comment]
same issue

**Reply by PickleOnAString** [reply]
restarting fixed it

**Comment by Syntex9000** [comment]
Restarting fixed it for me as well. 
VS 2022 17.5.1
Windows 11

**Comment by PatrikUgrin** [comment]
So restarting is the only solution. It is like car completely turning off on highway and stopping on the side of the road and restarting the car is the solution... On a highway...

**Reply by Syntex9000** [reply]
Stop being a butthurt melinneal. It's a system reboot. It takes like 20 seconds.

**Reply by jeremy-holovacs-sp** [reply]
> Stop being a butthurt melinneal. It's a system reboot. It takes like 20 seconds.

This is a violation of the [Code of Conduct](https://github.com/community/community/blob/c7f015fe0743f1ce66d6d4abe8115193e6008f6f/CODE_OF_CONDUCT.md). Please be respectful to other participants in this issue.

**Reply by PatrikUgrin** [reply]
> Stop being a butthurt melinneal. It's a system reboot. It takes like 20 seconds.

It is not a simple 20s restart. You must stop all debug sessions if you had them, lose all green-marks on code change which I usually orient myself with trough large codebase and if you launch multiple VS instances you must repeat for all instances where it doesn't work. Plus you must actually notice it not working since there is no immediate sign that the extension crashed.
It is very disruptive is what I am trying to get across.
On top of it, this is a released/updated Paid software.

**Comment by tilmann1** [comment]
@Brigapes Can you please update to VS 17.5.1 and version 1.77.0.0 of the Copilot VS extension and let us know whether you're still running into any issues? Thanks!

**Reply by PatrikUgrin** [reply]
Ok, i'll update and let you know, thanks

**Reply by PatrikUgrin** [reply]
After a few days it is not happening anymore.

**Reply by ayim** [reply]
@Brigapes We shipped a fix in Copilot v1.78.0.0 that fixes this issue (hopefully for good!)
Two easy steps to update:

- In Visual Studio, go to Extensions > Manage Extensions > Updates and click Update if available
- In the same dialog, click Installed, and make sure "Automatically update this extension" is checked for GitHub Copilot

**Comment by giovhy** [comment]
VS 17.5.3 + Copilot VS Extension 1.77.2.0 = "Preparing solution..."

Indefinitely, even if it is an extremely simple solution.

Screenshot: https://i.imgur.com/VcCnu8v.png

Language setup is Spanish from Spain.

The only way I found to use Visual Studio is: kill Visual Studio from Task Manager, then open Visual Studio without opening any project/solution, then disable the Copilot extension. After restarting VS, everything works. Except Copilot, of course.

Thank you, best wishes, and good luck.

PS: Running VS 2022 as administrador does not solve the issue.

**Reply by cateyelow** [reply]
so sad me too

**Reply by giovhy** [reply]
> VS 17.5.3 + Copilot VS Extension 1.77.2.0 = "Preparing solution..."
> 
> Indefinitely, even if it is an extremely simple solution.
> 
> Screenshot: https://i.imgur.com/VcCnu8v.png
> 
> Language setup is Spanish from Spain.
> 
> The only way I found to use Visual Studio is: kill Visual Studio from Task Manager, then open Visual Studio without opening any project/solution, then disable the Copilot extension. After restarting VS, everything works. Except Copilot, of course.
> 
> Thank you, best wishes, and good luck.
> 
> PS: Running VS 2022 as administrador does not solve the issue.

Hello again.

I hope the following information helps the extension developers. The problem I am having with GitHub Copilot is related to **network issues.**

My ISP is having problem with some routes, meaning that I cannot reliably connect to all destinations.
I then tried using a paid VPN. **While the paid VPN is connected, VS2022 + GitHub Copilot work just fine!**

Now that we know that this kind of problem can happen in relation to network issues, perhaps the developers of GitHub Copilot can improve the network related code to better control a possible network related issue to prevent these infinite hangouts. A more **graceful** way to handle this could be to inform the user that the extension may be experiencing network related issues.

For the other people who are having the same problems as me, I recommend trying a different internet connection or a good VPN. Maybe this will help you!

Thanks for reading. Best regards.

**Reply by ehasis** [reply]
I'm having exactly same issue. I'm paying for a service that does not work.

**Reply by ayim** [reply]
@ehasis @giovhy @cateyelow our team root caused this issue and have produced a fix, should release soon.
If you see issues in the future (i.e. slow suggestions, no suggestion, etc) please shoot me an email at ayim at github dot com so that we can root cause together and produce a fix as soon as possible

**Reply by ayim** [reply]
@ehasis @giovhy @cateyelow good news! We shipped a fix in Copilot v1.78.0.0 that fixes this issue.
Two easy steps to update:

- In Visual Studio, go to Extensions > Manage Extensions > Updates and click Update if available
- In the same dialog, click Installed, and make sure "Automatically update this extension" is checked for GitHub Copilot

**Comment by dragurimu** [comment]
![image](https://user-images.githubusercontent.com/38273104/232376540-ac1ae59b-4e98-40ed-9b2e-7c4c5edb6863.png)

30 years in a hint, github is horribly slow in vscode.
